### ISSUE TITLE
Core update page displays incorrect status after update

### ISSUE THREAD

**Issue by rica-carv** [e107inc/e107#1936]
Live site is currently a v1.4 upgraded to v2.x, but for some reason database update is stuck in core database structure, and github update gives a lot errors.....

![untitled 2](https://cloud.githubusercontent.com/assets/17130530/19448099/416fad00-9498-11e6-98fd-1a207326a8b2.jpg)

![untitled 1](https://cloud.githubusercontent.com/assets/17130530/19448125/5e85e7d8-9498-11e6-950a-6e6bd5cfe6c9.jpg)

Can anyone help to clean up this mess? Thanks....

**Comment by tgtje** [comment]
Seen it happen before, doing the sync again could help. (transfer from temp to dir looks failing sometimes > server glitches ???) (use ftp else).

All other related posts will suffer from this cause !! Visited selos, > example : forum has issues too... paths reveal old v1 forum files etc...; java errors  var showhide=new switchcontent > log (show/hide) etc.. Heavily adjusted theme construct, and some special plugins... hard to tell what all might have influence..
Do not forget (afaik) .tk is domain, but source is normally elsewere coupled.

**Comment by rica-carv** [comment]
@tgtje Donne new sync, same error. So i'm going with heavy artillery via FTP, as last resort.

Meanwhile i'm trying to make a V2.x of it without upgrading from V1.4, so therefore all those errors will be gonne (i hope...)
The theme is currently being (slowly) upgraded to v2.x....
The special plugins are also on the "to upgrade to v2.x" list, so....But as v1.4 they work on v2.x, but who knows when they'll start to fail....

**Comment by Moc** [comment]
Try to do a manual update. 

- Download the ZIP from Github, then upload and overwrite all files. 
- Run the database updates (in debug mode preferably, so we can see where it gets stuck if it does).

**Comment by rica-carv** [comment]
@Moc Manual update works, site is now updated to 2.1.2 (git), but github update still gives the same errors....

**Comment by CaMer0n** [comment]
These errors can often be ignored.  Consider it debug info. If files are unable to be unzipped then it's a server configuration issue.

**Comment by rica-carv** [comment]
@CaMer0n Did a database check, had some errors had to be corrected directly via Phpmyadmin, now it's ok, but v2 forum (database sucessfully upgraded) and links plugins gives me errors now....

**Comment by Moc** [comment]
Which errors?

**Comment by rica-carv** [comment]
@moc Only have this:
![1](https://cloud.githubusercontent.com/assets/17130530/19618084/f5a89356-9838-11e6-88c8-39538c7b2fb6.jpg)
![2](https://cloud.githubusercontent.com/assets/17130530/19618083/f5a6dd18-9838-11e6-9920-80c7bc19ebea.jpg)

Sorry, i forgot to create screens of the three errors.
It was errors due to discrepancies between fields structure and the sql files.
I guess it has something to do with my previous "attempt" to manually upgrade the database...

Anyway, now database seems ok, at least there's no admin UI updete message.
Got only some errors in a pair of plugins that might be due to my V1.4 theme template....
Still investigating....

**Comment by rica-carv** [comment]
@Moc I've pressed several times the core database strucuture button, and it would not update. After i've donne the database validity check, and see the three errors (which were also unable to correct from the admin UI interface, and whose i only managed to correct them using phpmyadmin), and after correcte them manually, only after all that, the core database button worked.
Now got it all right, at least there's no more database errors (yet...)...

**Comment by rica-carv** [comment]
@Moc I was able to reproduce two of the database validity check errors:
![untitled 1](https://cloud.githubusercontent.com/assets/17130530/19620104/5dbc4742-986d-11e6-827d-1fde2ec01397.jpg)
The third one was like the first on this screen capture, but on a different field of the same database.

I've had to manually repair them with phpmyadmin, and on the *_sql.php files:
![untitled 1](https://cloud.githubusercontent.com/assets/17130530/19620138/fff5f6e8-986d-11e6-8d3b-080e30817cd6.jpg)

Somehow, the space before the end comma gives this error.
After this was corrected on the php file, it was ok after the database validity check, anda also i was able to complete the core database update, and all is ok for now....

Meanwhile i keep investigating the templating issues....

**Comment by LaocheXe** [comment]
I will point out that an e107 v1.0.7 upgrade to v2.x gave me the core db update button alot, I fix my issue by editing the core back up preferences (i believe - been over a year), so either issue with the core preferences in the db or an old plugin. I also had to go in via phpmyadmin and remove the old plugins in the core db as well.

**Comment by rica-carv** [comment]
Indeed, althought it seems to be working, everytime i login in the admin ui area, it gives the message needed core database update, after i press the button, and for the session it doesn't show up again....

**Comment by LaocheXe** [comment]
@rica-carv try this, go to ur site, login, goto admin, do the steps to preform updates, after that go to your site preferences and change the site name, save it, go back to main admin area and see if the database still needs an update, if not then go back and change the name of the site back. Log out, log back in and see if the update button shows

**Comment by rica-carv** [comment]
What the heck, i must have gremlins on my site...
Today, first login, no database update message, all is clean.
Logged out, and logged in again, and again, no message....

Anyway, thanks for all the help...

**Comment by MikeyGMT** [comment]
Unfortunately I don't get passed the download stage. 
The UI reports "Couldn't download .zip file (DBLAN_118) 
Meanwhile I check my temp folder and sure enough the new zip file is there, alongside e107-master folder from 6 months ago. 
Added sleep(10); before if($result == false)
I could then see the file come in as 0644 and then get CHMODed to 0755
Uncommenting print_a($unarc); this returns 0.  
So it looks like it's not unzipping. 
In cPanel attempted to unzip, got this... so maybe the full file has not downloaded.

> End-of-central-directory signature not found.  Either this file is not
>  a zipfile, or it constitutes one disk of a multi-part archive.  In the
>  latter case the central directory and zipfile comment will be found on
>  the last disk(s) of this archive.

Sync e107.master.zip file size 7.4 MB, repeating several times I'm receiving anything from 6.7 MB to 7.55 MB master zip file. 
 
Downloaded file locally - 8.99 MB, uploaded to CentOS (commercial) host, same, 8.99 MB.

Increased sleep to 30s  then 45s and still the file is not complete.  

Hmmm ... check PHP Info
upload_max_filesize	 2 MB  
memcache.chunk_size	8192 
post_max_size	8M

Any ideas?

**Comment by MikeyGMT** [comment]
Aha
Added to file_class at line 455
```
// Check if any error occurred
if(curl_errno($cp))
{
    echo 'Curl error: ' . curl_error($cp).'<br />';
}
```
(It should be addError, of course).

A clue ... 

> Curl error: Operation timed out after 10000 milliseconds with 6778998 bytes received

OK so I added to line 443

`		curl_setopt($cp, CURLOPT_TIMEOUT, 20); `

And it worked! However it did finish with a empty white page.

**Comment by CaMer0n** [comment]
@rica-carv Please test by modifying core_sql.php. It should trigger an update request and normally after the update has run the "Update" button should now change to a tick. (check)

**Comment by CaMer0n** [comment]
Closing. Please open a new issue if something related to the above still exists.
